Fix for compile warnings[glusterd-utils.c]

pbhatia22 opened the pull request:
Warnings:
```
glusterd-utils.c:7709:13: note: ‘snprintf’ output between 29 and 3804 bytes into a destination of size 2048
             snprintf(msg, sizeof(msg),

glusterd-utils.c:7743:9: note: ‘snprintf’ output 86 or more bytes (assuming 4115) into a destination of size 2048
         snprintf(msg, sizeof(msg),
```

Warning arises as msg is having smaller size
so added extra check to remove warning

Updates: #1000
Change-Id: I65692d0a5ed39a06068519d49ca0eb11e15286a2
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 committed:
Fix for compile warnings[glusterd-utils.c]

Warning arises as msg is having smaller size
so added extra check to remove warning

Change-Id: I65692d0a5ed39a06068519d49ca0eb11e15286a2
Updates: #1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/granular-esh/cli.t

0 test(s) generated core 


8 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/01-georep-glusterd-tests.t
./tests/00-geo-rep/georep-basic-dr-rsync-arbiter.t
./tests/basic/afr/granular-esh/cli.t

4 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/2346/

pbhatia22 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/granular-esh/cli.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/00-geo-rep/georep-basic-dr-rsync.t
./tests/basic/afr/granular-esh/cli.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
https://build.gluster.org/job/gh_centos7-regression/2349/

pbhatia22 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/basic/afr/granular-esh/cli.t

0 test(s) generated core 


6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/basic/afr/granular-esh/cli.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
https://build.gluster.org/job/gh_centos7-regression/2353/

pbhatia22 committed:
Fix for compile warnings[glusterd-utils.c]

Warning arises as msg is having smaller size
so added extra check to remove warning

Updates: #1000
Change-Id: I65692d0a5ed39a06068519d49ca0eb11e15286a2
Signed-off-by: Preet <pbhatia@redhat.com>

pbhatia22 commented:
/run regression

amarts reviewed (commented):
lgtm

mohit84 commented on xlators/mgmt/glusterd/src/glusterd-utils.c:
```diff
@@ -7706,10 +7706,14 @@ glusterd_is_path_in_use(char *path, gf_boolean_t *in_use, char **op_errstr)
-                     "%s is already part of a "
-                     "volume",
-                     path);
+            int z = snprintf(msg, sizeof(msg),
```
I think we don't need to declare a new variable , we can use ret instead of using z.

pbhatia22 replied:
Using ret is causing a problem which i found out in previous run regression when we are assigning value to ret of snprintf when it is successful it can also take value other than zero but then in caller function of glusterd_is_path_in_use() will treat ret value as an error and proceed according to that

mohit84 replied:
ok
